packages/vitepress-plugin-enhanced-readabilities/src/client/components/LayoutSwitch.vue

```diff
@@ -101,13 +101,13 @@ watch(layoutMode, (val) => {
   if (!mounted.value)
     return
 
-  setClasses(val, true)
+  setClasses(val, true && !options.layoutSwitch?.disableAnimation)
   if (!supportedLayoutModes.includes(val))
     layoutMode.value = options.layoutSwitch?.defaultMode || LayoutMode.BothWidthAdjustable
 })
 
 watch(route, () => {
-  setClasses(layoutMode.value, true)
+  setClasses(layoutMode.value, true && !options.layoutSwitch?.disableAnimation)
```

Comment:
Why not just `!options.layoutSwitch?disableAnimation`?

```suggestion
  setClasses(layoutMode.value, !options.layoutSwitch?.disableAnimation)
```